### ISSUE TITLE
Feature Request: Open settings panel on after install

### ISSUE THREAD

**Issue by Samuel12321** [ShankarBUS/ModernFlyouts#146]
Store feedback;
Users aren't always aware of the settings menu. If this was opened just once at the first launch of the app, it would make them aware of the customisation available.

**Comment by Samuel12321** [comment]
As someone who just installed ModernFlyouts for the first time, I like the current behavior. **Had it been hidden by default I probably never would've known any of those controls existed.**

_Originally posted by @justintaddei in https://github.com/ShankarBUS/ModernFlyouts/issues/125#issuecomment-716344621_

**Comment by ShankarBUS** [comment]
Is this necessary?

I don't mind doing it but wouldn't the users be annoyed by it?

Do you want me to make a OOBE?

**Comment by Samuel12321** [comment]
As long as it only appears on the first time the app is run, i don't think it would be annoying.
Others feedback would be useful.

**Comment by ShankarBUS** [comment]
So what about existing users?

There's no way to distinguish existing users and new users currently. I never thought of this scenario. But it wouldn't be problematic in the next version.

However all users will get the OOBE during the first implementation.

Is this something the users hate?

**Comment by Samuel12321** [comment]
i think it should be fine as long as it is explained that it will only appear once.

**Comment by ShankarBUS** [comment]
Hey pal,

I also plan to include way to enable certain experimental features only for beta users which won't be available for normal users.

This could be useful right?

**Comment by Samuel12321** [comment]
Should we do this as a setting in the settings menu. eg "enable beta features"
Or use the beta flight ring that we have and push those updates to that ring.

**Comment by justintaddei** [comment]
Me again :)

I didn't have any trouble finding the settings page since the settings icon is pretty obvious.  
This one:
![image](https://user-images.githubusercontent.com/16705594/97525567-18a0ee00-1975-11eb-8a60-43e8a487548c.png)

I don't think it would've bothered me if the settings page had opened on the first launch. Perhaps a less obtrusive welcome screen that asks if the user would like to customize the behavior or sick with the defaults would be a better option?

**Comment by ShankarBUS** [comment]
Use the beta flight ring.

I thought it was not being used so I came to this decision

**Comment by Samuel12321** [comment]
@justintaddei Thanks for clarifying, also got store feedback regarding not knowing about the settings page and assumed this is what you meant.

**Comment by Samuel12321** [comment]
> Use the beta flight ring.
> I thought it was not being used so I came to this decision

It is being used, the last release 0.7.5 was in the beta ring for 2 days to monitor for bugs before i released it to public.

**Comment by ShankarBUS** [comment]
@justintaddei,

> Perhaps a less obtrusive welcome screen that asks if the user would like to customize the behavior or sick with the defaults would be a better option?

That's what I mentioned as OOBE (out of the box experience).

There will be a welcome screen explaining the features of this app and asking the users for customization.

**Comment by ShankarBUS** [comment]
@Samuel12321, Oh! I didn't know that.

That's great!

**Comment by Samuel12321** [comment]
So how would we do this, would we have a fork with beta features with its own releases that go to the beta ring, that get merged into the main branch, or as we currently are with one branch and each release just goes to beta ring for a few days before public release.

**Comment by ShankarBUS** [comment]
No need for that we just have to add **"-beta"** to the app version on the *.csproj file.

I can then filter out features based on the app version.

If the version contains "beta" it will enable certain features.

We can remove suffix once the app becomes stable and release the stable version to store

**Comment by Samuel12321** [comment]
I wasn't aware that version in csproj could contain text. Sounds good to me.
FYI:
We currently have 3 release rings. 
- (Internal beta) - just us, have been using this for a while to test update mechanism and check for distribution issues.
- (Beta) - everyone who signs up for the beta
- (public) - everyone

**Comment by Samuel12321** [comment]
just working out future releases. How is the 0.8 stuff going?

**Comment by ShankarBUS** [comment]
@Samuel12321, I'm fiddling with the settings window.

I'm nearly close to fixing the settings UI lag.

All the other planned stuff has been done. Just this one is left.

It may take me a couple of days to wrap it up for beta users together with this new OOBE and feature previews.

**Comment by justintaddei** [comment]
@ShankarBUS 

> That's what I mentioned as OOBE (out of the box experience).
> 
> There will be a welcome screen explaining the features of this app and asking the users for customization.

My bad. I don't know why I didn't pick up on that. Sounds perfect!

**Comment by Samuel12321** [comment]
can you push what you currently have.

So do you want to have OOBE and beta specific stuff for 0.8.0?

**Comment by ShankarBUS** [comment]
@Samuel12321, It's messy over here. I implemented all the planned features at once. I'll upload all of the settings window features at once.

Should have done separate commits for each feature.

**Comment by Samuel12321** [comment]
Just planning releases did you want OOBE and beta specific stuff for 0.8.0 or 0.8.5?

**Comment by ShankarBUS** [comment]
We can add OOBE and beta-specific features in v0.8.5

Let's fix the settings UI bug first.
What should be the first beta-specific feature?

**Comment by Samuel12321** [comment]
> We can add OOBE and beta-specific features in v0.8.5
> Let's fix the settings UI bug first.

Agreed

> What should be the first beta-specific feature?

Don't know?
> Should they be in a separate page in settings? So that beta users can know which features are experimental and enable them on their own.

Yes

**Comment by Samuel12321** [comment]
perhaps they should be by default on, so that we can get feedback if they are causing crashes or problems.

**Comment by Cyberdroid1** [comment]
> No need for that we just have to add **"-beta"** to the app version on the *.csproj file.
> 
> Example : 0.8.0-beta

What about naming it as 0.8.0 Preview?   Many apps name it like that 😁

**Comment by Cyberdroid1** [comment]
For the OOBE, have a look at how EarTrumpet does it.

**Comment by ShankarBUS** [comment]
> What about naming it as 0.8.0 Preview?

There are many types of pre-releases such as **alpha, beta** and **preview**. **"0.8.0-preview"** would do a similar job of what **"beta"** does. While the suffix **"Preview"** is predominantly used for denoting pre-release version of libraries (take .NET 5 or even ModernWpf as an example), the suffix **"Alpha/Beta"** is used for denoting software pre-releases. Most of the apps on **Google Play Store/MS Store** uses **beta** suffix for pre-release builds.

https://en.wikipedia.org/wiki/Software_release_life_cycle
https://stackoverflow.com/a/736448, **"Preview"** will sound confusing for most users. **"Beta"** is more commonly used.

Even MS Store provide a channel only for **"Beta"** and **"Stable"** builds. So, I thought the suffix **"-beta"** would be suitable.

> Many apps name it like that 😁

This statement is absurd. Even **PowerToys** don't use the term **"Preview"** they just use **"-Experimental".**

**Comment by Cyberdroid1** [comment]
@ShankarBUS 
![image](https://user-images.githubusercontent.com/51435948/97532934-5f2a3300-19dd-11eb-95d0-057f66ce78a1.png)

**Comment by Samuel12321** [comment]
i agree with @ShankarBUS "beta"

**Comment by Samuel12321** [comment]
the settings tab will be called experimental

**Comment by Cyberdroid1** [comment]
Well, its your call at the end, 😁. As you wish.

**Comment by ShankarBUS** [comment]
@Cyberdroid1,

![image](https://user-images.githubusercontent.com/63704247/97533347-0effa080-19de-11eb-8d23-2e8d4031a633.png)
![image](https://user-images.githubusercontent.com/63704247/97533384-1de65300-19de-11eb-8bf6-c6956d40d356.png)

Ooooh! I forgot that 😅😁. @Samuel12321 your thoughts?

**Comment by Samuel12321** [comment]
very good points, guess preview wins.

**Comment by ShankarBUS** [comment]
Oh wait,

The term **"Preview"** in this context denotes that all builds from **stable** to **experimental** (including alpha & beta) are not public ready.
This term doesn't denote the build channel. The app itself is in preview mode.

My point wins.

**Comment by Samuel12321** [comment]
for the internal numbering i still think it should be "beta", but for exterior visible labels "preview"

**Comment by ShankarBUS** [comment]
> for the internal numbering i still think it should be "beta", but for exterior visible labels "preview"

No, we don't need to. We should have added the term "Preview" to the app's name at the beginning.
Any software before attaining version **"1.0.0.0"** is called **"Preview"**.

So, our app (currently v0.7.8), PowerToys (currently v0.24.0) and Files UWP (currently v0.17.0) are all in **Preview**.

We are all under "1.0.0.0". After attaining v1.0 the **"Preview"** tag would be removed.
We have to add "Preview" to all our build channels (beta & stable).

**Comment by Cyberdroid1** [comment]
The app name should not tell that we are using Beta version, the About Page should contain it.

**Comment by ShankarBUS** [comment]
Yup!
@Samuel12321, could you change our app & package display name to **"ModernFlyouts (Preview)"**?

**Comment by Cyberdroid1** [comment]
Yup, makes sense. 😁

**Comment by Samuel12321** [comment]
Will do with the next release. 😉

**Comment by thomasthebear** [comment]
> _So what about existing users?_
> 
> _There's no way to distinguish existing users and new users currently. I never thought of this scenario. But it wouldn't be problematic in the next version._
> 
> _However all users will get the OOBE during the first implementation._
> 
> _Is this something the users hate?_    
    
***    
     
Unsure if this has already been addressed and _my apologies_ for commenting on a Issue almost two weeks after the fact, but from a very basic level of implementing the ability for the software to distinguish between **First Time/New Users** and **Existing Users** would be to have the software show the Settings Dialog _(or even a New User/Welcome Pane)_ on startup, by default. This allows the software to always show the designated panel on first run, without fail. Once the Dialog/Panel is opened, either on the Initial Installation or Initial Startup, generate a basic text file _(or similar properties file)_ in its directory with a simple line of something similar to `  out-of-box.experience = false  `, `  startup.settings-panel.show = false  `  or honestly **anything** of your desired preference. Run a quick check of that line in the properties file to return `  true  ` or `  false  ` on start up while having a small section of internal coding that would designate what it should show based upon the results of said check.     
    
     
**That being said,** this is coming from someone with limited experience in terms of writing my own software or contributing to existing projects. However, I do have a relatively good understanding of most programming languages and their overall uses & implementation, _or so I'd like to think_.  Hopefully this helps, even if in the slightest, to put yourself on the right track for implementation of an "**_Out of Box Experience_**" for New Users, if you're planning to do so eventually.

**Comment by ShankarBUS** [comment]
@thomasthebear, thanks for you suggestion.

That's exactly what I planned! You're suggesting to use app data to store if it's first run or not. Which is how it should be done!

But since the data is not already present in the data list, a new entry with `true` as value would be created.

As the entry is not present in existing installations, existing users will also get to see the OOBE only once.

They won't appear again.

I guess you misunderstood my statement 😄.

By existing users/installation I meant users that already know about the app and have ModernFlyouts installed and been using it for a while.

I hope we're clear 😁.

**Comment by ShankarBUS** [comment]
@Samuel12321,

I'm going to skip this and start the work for v0.9.
We can add this to v1.0. I have to use my free time effectively. The OOBE stuff is easy but time consuming (UI designing, localization, screenshots and etc). You could even do that once you fully learn C#, .NET and WPF. I need to code as much as possible within this month and the next. We have a lot of stuff to implement.

I need a small help from you. We need help from a **UI/UX designer**. Could you search for anyone and ask them to design mockups for our app ASAP?

**Comment by Cyberdroid1** [comment]
Hey, what about Sharlee? He had designed the icon for the Dopamine 3 app.

**Comment by Samuel12321** [comment]
Will see if i can find anyone.
@ShankarBUS I was thinking it may be a good idea to talk to @riverar and the Ear trumpet team to see whether they would be interested in a possible project merge for #164, seeing as we are both doing similar things, (ours being a bit broader), it sort of makes sense to pool resources. - They may have someone in their team who can help with ui/ux design.

**Comment by zeealeid** [comment]
Hm, maybe when the app gets opened for the first time, the user gets a pop up window (like that 'What's new' window thing in new Insider builds) showing the features of the app and an option for accessing the settings area. Would create some mock-ups to explain it better.

**Comment by ShankarBUS** [comment]
@zeealeid,

> Would create some mock-ups to explain it better.

That's what I need exactly!

Great! ❤️

**Comment by zeealeid** [comment]
> Hm, maybe when the app gets opened for the first time, the user gets a pop up window (like that 'What's new' window thing in new Insider builds) showing the features of the app and an option for accessing the settings area. Would create some mock-ups to explain it better.

Ok so I was saying that when the app opens for the first time, the user gets greeted with this pop-up (heavily inspired by Tips app). So this is what I've come up with


![Welcome](https://user-images.githubusercontent.com/54409865/103644944-74d35000-4f6c-11eb-9943-08631105a8a4.png)

![Music](https://user-images.githubusercontent.com/54409865/103645292-0642c200-4f6d-11eb-803a-71636fbba3ee.png)

![Other flyouts](https://user-images.githubusercontent.com/54409865/103645316-0d69d000-4f6d-11eb-9c29-40287c1c7a5d.png)

![Settings](https://user-images.githubusercontent.com/54409865/103645318-0d69d000-4f6d-11eb-84a1-caf8c1138750.png)

Let me know your thoughts on this!

**Comment by Cyberdroid1** [comment]
NGL, the design is lip smacking!
Like the privacy policy and settings button addition too! 

But we would need someone to implement it, as @ShankarBUS does not have time to implement it.

**Comment by zeealeid** [comment]
> NGL, the design is lip smacking!
> 
> Like the privacy policy and settings button addition too! 
> 
> 
> 
> But we would need help of someone to implement it, as @ShankarBUS does not have time to implement it.

Glad you like it! Maybe this could be implemented in a later release?

**Comment by ShankarBUS** [comment]
@zeealeid,


Wowie! The OOBE UI looks better than the app itself 🤣.
That's too better!

As @Cyberdroid1, I don't have much time.

But I'll try to sneak it in somehow.

I'll at least make the process easier for future contributors.

Thank you so much pal!

**Comment by zeealeid** [comment]
> @zeealeid,
> 
> 
> Wowie! The OOBE UI looks better than the app itself 🤣.
> That's too better!
> 
> As @Cyberdroid1, I don't have much time.
> 
> But I'll try to sneak it in somehow.
> 
> I'll at least make the process easier for future contributors.
> 
> Thank you so much pal!

You're welcome! Glad you like it
